### DISCUSSION TITLE
How to add own shapes to library

### DISCUSSION THREAD

**Discussion by emulite** [SVG-Edit/svgedit#977] [Q&A]
Have worked out how to create the appropriate category json file to hold the path information for new symbols, and how to add that option to the menu that pops up when you tap on the shapes library icon (via index.json and locale/en.js) but falling down on populating the image-lib that seems to hold the thumbnails for the shapes within the json for the new category.
Looking at the thumbnails, they seem to be a svg-xml hybrid coded base64. But which script produces them from what reference>
Any hints please?

**Comment by emulite** [comment]
Spotted my own error - didn't use  'data' at the top level of the json and substituted another name

**Comment by UsingChipNewInstructionToAvoidBranch** [comment]
I am wondering if I have enough little icon library ,whether I can create a can-use indoor map .
